dspace-api/src/main/java/org/dspace/discovery/SolrServiceUnDiscoverableItemsPlugin.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.discovery;
+
+import static org.apache.logging.log4j.LogManager.getLogger;
+
+import java.sql.SQLException;
+
+import org.apache.logging.log4j.Logger;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.core.Context;
+import org.dspace.core.LogManager;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class SolrServiceUnDiscoverableItemsPlugin implements SolrServiceSearchPlugin {
```

Comment:
Do we need this additional filter?
Search results will by default verify permissions based on SolrServiceResourceRestrictionPlugin, which should already imply only admins can find withdrawn or non-discoverable items

Reply:
Thanks @benbosman , I'm sure you are right about this.  I'll first verify that the discovery permissions are in place where you indicate and then remove this plugin.  When the Jira issue was written a separate plugin was suggested and this class simply implemented the requirement without verifying that it is needed.

Reply:
Yes, the plugin checks read permissions for all non-system admins.  That presumably makes unnecessary any additional filter to exclude withdrawn and non-discoverable items.  I can also remove filters from the discovery.xml configurations below (that I had tentatively added).

If the pattern holds, this PR will entail only a new discovery.xml configuration and ITs!

Reply:
Although it's not needed for withdrawn items, a plugin appears to be needed for private items.

Reply:
you're right, thinks for this update